### ISSUE TITLE
Pandas 2.1.0 and mfi indicator 

### ISSUE THREAD

**Issue by zakcali** [twopirllc/pandas-ta#731]
**Which version are you running? The lastest version is on Github. Pip is for major releases.**
pandas-ta 0.3.14b
python 3.12
pandas 2.1.0
**Do you have _TA Lib_ also installed in your environment?**
No

**Have you tried the _development_ version? Did it resolve the issue?**
no, because is not compatible with python 3.12

**Describe the bug**
[A clear and concise description of what the bug is.](futurewarning: setting an item of incompatible dtype is deprecated and will raise in a future error of pandas)

**To Reproduce**
```python
df["MFI"] = ta.mfi(high=df["High"], low=df["Low"], close=df["Close"], volume=df["Volume"], length=14, talib=False)
```

**Expected behavior**
pandas 2.0.3 doesn't give that warning

**Screenshots**
[Reason is updated pandas behaviour](https://pandas.pydata.org/pdeps/0006-ban-upcasting.html)

**Additional context**
I find a function from: https://gist.github.com/quantra-go-algo/d3ced509e55ceb15eb0a6bd283e6853f#file-ti_mfi-py 
and fine tuned with chatgpt to execute faster than original code and pandas-ta:
I couldn't correct the pandas-ta bug, so I embedded below code to my python program

```python
def mfi(high, low, close, volume, n):
    typical_price = (high + low + close) / 3
    money_flow = typical_price * volume
    mf_sign = np.where(typical_price > typical_price.shift(1), 1, -1)
    signed_mf = money_flow * mf_sign

    # Calculate gain and loss using vectorized operations
    positive_mf = np.where(signed_mf > 0, signed_mf, 0)
    negative_mf = np.where(signed_mf < 0, -signed_mf, 0)

    mf_avg_gain = pd.Series(positive_mf).rolling(n, min_periods=1).sum()
    mf_avg_loss = pd.Series(negative_mf).rolling(n, min_periods=1).sum()

    return (100 - 100 / (1 + mf_avg_gain / mf_avg_loss)).to_numpy()
```
  
Thanks for using Pandas TA!

**Comment by twopirllc** [comment]
Hello @zakcali,

> I couldn't correct the pandas-ta bug, so I embedded below code to my python program

Nice! Glad you find a work around and shared the code. 😎

Like many indicators, including _mfi_, are in _need_ of an update. So this really helps! Thanks 😎

Kind Regards,
KJ

**Comment by zakcali** [comment]
this one run  faster, by using numpy instead of pandas on my 376 item xu100 market

# Calculate money flow index
def calculate_mfi(high, low, close, volume, period):
    typical_price = (high + low + close) / 3
    money_flow = typical_price * volume
    mf_sign = np.where(typical_price > np.roll(typical_price, shift=1), 1, -1)
    signed_mf = money_flow * mf_sign

    # Calculate gain and loss using vectorized operations
    positive_mf = np.maximum(signed_mf, 0)
    negative_mf = np.maximum(-signed_mf, 0)

    mf_avg_gain = np.convolve(positive_mf, np.ones(period), mode='full')[:len(positive_mf)] / period
    mf_avg_loss = np.convolve(negative_mf, np.ones(period), mode='full')[:len(negative_mf)] / period

    epsilon = 1e-10  # Small epsilon value to avoid division by zero
    mfi = 100 - 100 / (1 + mf_avg_gain / (mf_avg_loss + epsilon))
    return mfi

df["MFI"] = calculate_mfi(high=df["High"].values, low=df["Low"].values, close=df["Close"], volume=df["Volume"].values, period=MFI_PERIOD)

**Comment by twopirllc** [comment]
Hello @zakcali 

Thanks for submitting a faster version. It is now available on the _development_ branch.

KJ

**Comment by twopirllc** [comment]
Hello @carissssa,

A new main release will occur after outstanding PRs are completed and some other bugs and maintenance issues have been taken care of. When, however, is unknowable at this time. 😕 

> i cannot use dev branches and this is blocking pandas upgrade

What is preventing the use of the _dev_ branch?

Kind Regards,
KJ
